src/libraries/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -1053,30 +1053,41 @@ private void CreateThisFromUri(Uri otherUri)
         {
             DebugAssertInCtor();
 
-            // Clone the other URI but develop own UriInfo member
-            _info = null!;
-
             _flags = otherUri._flags;
-            if (InFact(Flags.MinimalUriInfoSet))
+
+            if (InFact(Flags.AllUriInfoSet))
+            {
+                // We can share it now without mutation concern, for since AllUriInfoSet it is immutable.
+                _info = otherUri._info;
+            }
+            else
             {
```

Comment:
```suggestion
            {
                Debug.Assert(!InFact(Flags.HasUnicode));
```

Reply:
I have tried and several tests failed. When I think about it now, I am not sure if we have to assert it is not unicode. As for example here: https://github.com/dotnet/runtime/blob/03b2d3d73250a210109d803a65299c0e9477bfff/src/libraries/System.Private.Uri/tests/FunctionalTests/IriRelativeFileResolutionTest.cs#L253-L263
it will crash at line 261 when `startUri` contains unicode. `MakeRelativeUri` does not mark it as `AllInfoUriSet`.
I thinki about two ways, either `EnsureParseRemaining` which might have unexpected perf cost, or `Debug.Assert(!InFact(Flags.HasUnicode) || otherUri.IsNotAbsoluteUri);`